### ISSUE TITLE
ES download is sometimes flaky

### ISSUE THREAD

**Issue by cachedout** [elastic/elastic-stack-testing#307]
This problem was observed in this test:

https://internal-ci.elastic.co/job/elastic+estf-monitoring-snapshots+7.3+multijob-kibana/49/console

```
13:24:47 included: /var/lib/jenkins/workspace/elastic+estf-monitoring-snapshots+7.3+multijob-kibana/ansible/roles/xpack_elasticsearch/tasks/linux/xpack_elasticsearch_set_passwords.yml for aithost => (item={'action': 'xpack_elasticsearch_set_passwords', 'parent': '', 'args': {}})
13:24:47 
13:24:47 TASK [xpack_elasticsearch : Download elasticsearch tar package] ****************
13:24:50 changed: [aithost] => {
13:24:50     "changed": true,
13:24:50     "checksum_dest": null,
13:24:50     "checksum_src": "7ddffa39e9bd9de04d50c61625cae8c929454e0d",
13:24:50     "dest": "/tmp/elasticsearch-7.3.1-SNAPSHOT-linux-x86_64.tar.gz",
13:24:50     "gid": 1000,
13:24:50     "group": "vagrant",
13:24:50     "md5sum": "9c878d4ab62b2a2b411a05d46c41ecf6",
13:24:50     "mode": "0664",
13:24:50     "owner": "vagrant",
13:24:50     "size": 122133066,
13:24:50     "src": "/home/vagrant/.ansible/tmp/ansible-tmp-1566307487.6015034-52301487000183/tmp28mGkN",
13:24:50     "state": "file",
13:24:50     "status_code": 200,
13:24:50     "uid": 1000,
13:24:50     "url": "https://snapshots.elastic.co/7.3.1-426d4738/downloads/elasticsearch/elasticsearch-7.3.1-SNAPSHOT-linux-x86_64.tar.gz"
13:24:50 }
13:24:50 
13:24:50 MSG:
13:24:50 
13:24:50 OK (285071911 bytes)
13:24:50 
13:24:50 TASK [xpack_elasticsearch : Install elasticsearch tar package] *****************
13:24:52 fatal: [aithost]: FAILED! => {
13:24:52     "changed": false
13:24:52 }
13:24:52 
13:24:52 MSG:
13:24:52 
13:24:52 Failed to find handler for "/tmp/elasticsearch-7.3.1-SNAPSHOT-linux-x86_64.tar.gz". Make sure the required command to extract the file is installed. Command "/bin/tar" could not handle archive. Command "/usr/bin/unzip" could not handle archive.
13:24:52 	to retry, use: --limit @/var/lib/jenkins/workspace/elastic+estf-monitoring-snapshots+7.3+multijob-kibana/playbooks/monitoring/kibana/docs_parity.retry
```

Note that the download seemed as though it only took three seconds which seems unrealistic, especially in light of a comparison to [another test](https://internal-ci.elastic.co/job/elastic+estf-monitoring-snapshots+7.3+multijob-kibana/50/consoleFull) shows it taking around 42 seconds.

Assuming that the file wasn't totally downloaded, it also seems odd that the checksum passed and it's curious that we got a 200 from the web server as well. 

It would be nice if we could retry the download if the unarchive task fails but I'm not sure that Ansible supports this sort of on-fail behavior. Perhaps @liza-mae has run into this before?

cc: @ycombinator

**Comment by liza-mae** [comment]
@cachedout I have seen this before, a handful of times, I agree we should take a deeper look at what might be the root cause and how to better handle it in our ansible roles.

**Comment by liza-mae** [comment]
This should be fixed with my latest PR: #373 - I tried only retry with delay before but it did not work, since Ansible won't retry on failure, however it should retry when using until which is what I added.  Since I can't reproduce the failure locally - we will just have to wait and see if it still continues to fail on CI and I will adjust it from there if it does.

**Comment by cachedout** [comment]
@liza-mae Thanks for the update. I am going to set this as closed and if it re-occurs we can just open it back up. Thanks for your work on this!
